marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs

```diff
@@ -266,38 +266,58 @@ class Flow {
     // declareUpdate({explicitCommit: true});
     let flowInstance = this;
     let processor = flowInstance.step.getStepProcessor(flowInstance, step.stepDefinitionName, step.stepDefinitionType);
-    if(!(processor && processor.run)) {
+    if (!(processor && processor.run)) {
       let errorMsq = `Could not find main() function for process ${step.stepDefinitionType}/${step.stepDefinitionName} for step # ${stepNumber} for the flow: ${flowName} could not be found.`;
       flowInstance.datahub.debug.log({message: errorMsq, type: 'error'});
       throw Error(errorMsq);
     }
 
-      let hookOperation = function() {};
-      let hook = step.customHook;
-      if(!hook || !hook.module){
-        hook = processor.customHook;
-      }
-      if (hook && hook.module) {
-        // Include all of the step context in the parameters for the custom hook to make use of
-        let parameters = Object.assign({uris, content, options, flowName, stepNumber, step}, hook.parameters);
-        hookOperation = function () {
-          flowInstance.datahub.hubUtils.invoke(
-            hook.module,
-            parameters,
-            hook.user || xdmp.getCurrentUser(),
-            hook.runBefore ? flowInstance.globalContext.sourceDatabase : flowInstance.globalContext.targetDatabase
-          );
-        }
+    let hookOperation = function () {};
+    let hook = step.customHook;
+    if (!hook || !hook.module) {
+      hook = processor.customHook;
+    }
+    if (hook && hook.module) {
+      // Include all of the step context in the parameters for the custom hook to make use of
+      let parameters = Object.assign({uris, content, options, flowName, stepNumber, step}, hook.parameters);
+      hookOperation = function () {
+        flowInstance.datahub.hubUtils.invoke(
+          hook.module,
+          parameters,
+          hook.user || xdmp.getCurrentUser(),
+          hook.database || hook.runBefore ? flowInstance.globalContext.sourceDatabase : flowInstance.globalContext.targetDatabase
```

Comment:
I believe you need to force your expected order of operations here. 
What this does:
```javascript
(hook.database || hook.runBefore) ? flowInstance.globalContext.sourceDatabase : flowInstance.globalContext.targetDatabase
```
What you want:
```javascript
hook.database || (hook.runBefore ? flowInstance.globalContext.sourceDatabase : flowInstance.globalContext.targetDatabase)
```

Reply:
Fixed via https://github.com/marklogic/marklogic-data-hub/pull/2658/commits/9301cff9825b993ff4ade51acfc28f426179796f